packages/slate-react/src/components/inline.js

```diff
@@ -12,6 +15,58 @@ export default class Inline extends React.Component {
   render() {
     const { editor, node, parent } = this.props
 
+    const decorations = node.getDecorations(editor)
+    const children = []
+
+    if (
+      node.nodes.length > 1 ||
+      (node.nodes.length === 1 && node.nodes.first().text !== ' ')
+    ) {
+      for (const child of node.nodes) {
+        const i = children.length
```

Comment:
Hmmmm, I wonder if this is a bug or not.  I guess it depends on what the transpiled code looks like.  In the old days `i` definitely would have been shared between all iterations of the loop.  Even if we did fix it, I bet this code just doesn't work.

For example, what happens if the first node unmounts.  The other nodes won't remount, which means we'd have 0 -> null, 1 -> valid ref, except the "1" is now really the first node, it just never remounted.  There's no way this works.

Regardless though, not your bug, just noticed it now as I was reading.